ros2sim/parsers/sim_executor.py

```
@@ -0,0 +1,110 @@
+    # TODO: Modify get obs() to only return ints
+    values, labels = self.env.get_obs()
+    values = np.degrees(values)
+    for i, deg in enumerate(values):
+      if deg < 0:
+        values[i] += 360
+
+    ## This is important to preserve the exact values of joint values.
+    # Converting frop degrees to radians and back to degrees can cause some floating point
+    # differences to exist. This essentially gets rid of them.
+    values = np.round(values)
+
+    observation_packet = JointInformation()
+    for i, label in enumerate(self.obs_list[0:3]):
+      observation_packet.set_theta_value(label, int(values[labels.index(label)]))
+    for i, label in enumerate(self.obs_list[3:]):
+      observation_packet.set_joint_value_from_state(
+        self.joint_ordering_to_read_state[label], int(values[labels.index(label)])
+      )
+    observation_packet.at_goal = np.allclose(values[9:], np.degrees(self.current_goal),
+      rtol=0, atol=self.acceptable_tolerance)
+    return observation_packet
+
+  def action(self, packet: JointInformation):
+    """Apply the action to the robot. The values provided are in degrees. This is converted to radians
+    in the range -pi to pi to be used with gym solo.
+
+    Args:
+      packet (JointInformation): A joint information packet that contains the joint values to be executed
+    """
+    try:     
+      action_deg = []
+      for joint in self.env.joint_ordering:
+        # We want to ignore all the ankle joints
+        if joint[-5:] != "ANKLE":
+          action_deg.append(
+            packet.get_joint_value_from_state(
+              self.joint_ordering_to_read_state[joint]
+            )
+          )
+        else:
+          action_deg.append(0)
+      
+      # If the number of joint values provided has a mismatch with the number of joints in teh solo8 model
+      if len(action_deg) != len(self.env.joint_ordering):
+        raise ValueError
```

Do we want to be suppressing this exception? In other words, should we remove the try/except so that the program just exists? 

Not sure what the intended behavior is, so your call.